### PR TITLE
Add release note for v2.1

### PR DESCRIPTION
**Pull request by kvartet**
~TODO: update release date~

~Please update retiarii roadmap.~

### CODE REVIEW

**Comment by QuanluZhang** [comment]
> TODO: update release date
> 
> Please update retiarii roadmap.

@kvartet retiarii roadmap updated, please copy the completed items to this release note, thanks

File: docs/en_US/Release.rst [hunk 1]
```diff
@@ -5,6 +5,69 @@
...
+* Improve NAS 2.0 (Retiarii) Framework (Alpha release)
+
+  * `Feature roadmap <https://github.com/microsoft/nni/issues/3301>`__
+  * `Related issues and pull requests <https://github.com/microsoft/nni/issues?q=label%3Aretiarii-v2.1>`__
+  * Documentation improvement (#3404 #3432)
+
+Training service
+""""""""""""""""
+
+* Support shared storage for reuse mode (#3354)
```
Comment (QuanluZhang) [inline docs/en_US/Release.rst:26]: do we have official name for "reuse mode"? "reuse mode" seems too adhoc...

File: docs/en_US/Release.rst [hunk 2]
```diff
@@ -5,6 +5,69 @@
...
+* Support local platform for Windows in hybrid training service (#3353)
+* Remove PAIYarn training service (#3327)
+* Add recently-idle environment scheduler in reuse mode (#3375)
```
Comment (QuanluZhang) [inline docs/en_US/Release.rst:29]: -> Add "recently-idle" scheduling algorithm and set it as default scheduling algorithm in reuse mode

File: docs/en_US/Release.rst [hunk 3]
```diff
@@ -5,6 +5,69 @@
...
+* Rename ``preCommand`` to ``pythonPath`` and change command lines as serial in remote training service (#3284 #3410)
```
Comment (QuanluZhang) [inline docs/en_US/Release.rst:30]: don't understand...

File: docs/en_US/Release.rst [hunk 4]
```diff
@@ -5,6 +5,69 @@
...
+
+nnictl & nni.experiment
+"""""""""""""""""""""""
+
+* Migrate nnicli to new Python API ``nni.experiment`` (#3334)
+* Support resuming an experiment launched from Python by nnictl (#3309)
```
Comment (QuanluZhang) [inline docs/en_US/Release.rst:36]: seems like a bug...

File: docs/en_US/Release.rst [hunk 5]
```diff
@@ -5,6 +5,69 @@
...
+* Align ``nni.experiment`` tuner behavior with nnictl (#3419)
```
Comment (QuanluZhang) [inline docs/en_US/Release.rst:37]: -> Refactor the way of specifying tuner in experiment Python API (``nni.experiment``), more aligned with nnictl

File: docs/en_US/Release.rst [hunk 6]
```diff
@@ -5,6 +5,69 @@
...
+* Add Python API examples and documentation (#3396)
+* Add SECURITY doc (#3358)
+* Add 'What's NEW!' section in README (#3395) 
+* Update English contributing doc (#3398, thanks external contributor @Yongxuanzhang)
+
+Bug fixes
+^^^^^^^^^
+
+* Fix AML outputs path and python process not killed (#3321)
+* Fix env folder bug in reuse mode (#3374)
```
Comment (QuanluZhang) [inline docs/en_US/Release.rst:60]: not like a bug, but a feature

**Comment by kvartet** [comment]
> > TODO: update release date
> > Please update retiarii roadmap.
> 
> @kvartet retiarii roadmap updated, please copy the completed items to this release note, thanks

Updated all.

File: docs/en_US/Release.rst (hunk 1 above)
Comment (kvartet) [reply docs/en_US/Release.rst:27]: The document is like [this](https://nni.readthedocs.io/en/latest/reference/experiment_config.html?#reusemode), maybe a new name is needed?

File: docs/en_US/Release.rst (hunk 3 above)
Comment (kvartet) [reply docs/en_US/Release.rst:31]: I don't understand either, maybe @J-shang can provide a suggestion?

File: docs/en_US/Release.rst (hunk 6 above)
Comment (kvartet) [reply docs/en_US/Release.rst:61]: The PR description is 

1. Fix local reuse mode script error
~2. Fix env working folder, remove duplicated `env/{envId}`~
2. Fix reuse mode temp folder, remove duplicated `{experimentId}`

If it is a feature, I will modify it immediately.

File: docs/en_US/Release.rst [hunk 7]
```diff
@@ -5,6 +5,69 @@
...
+
+  * `Feature roadmap <https://github.com/microsoft/nni/issues/3301>`__
+  * `Related issues and pull requests <https://github.com/microsoft/nni/issues?q=label%3Aretiarii-v2.1>`__
+  * Documentation improvement (#3404 #3432)
+
+Training service
+""""""""""""""""
+
+* Support shared storage for reuse mode (#3354)
+* Support local platform for Windows in hybrid training service (#3353)
```
[the remark this answers is not in the document]
Comment (kvartet) [reply docs/en_US/Release.rst]: And the [doc](https://nni.readthedocs.io/en/latest/TrainingService/HybridMode.html) needs to be updated.

File: docs/en_US/Release.rst (hunk 3 above)
Comment (J-shang) [reply docs/en_US/Release.rst:31]: Maybe "Deprecated ``preCommand`` and enable ``pythonPath`` for remote training service." ?

File: docs/en_US/Release.rst [hunk 8]
```diff
@@ -5,6 +5,70 @@
...
+  * Support Random/Grid/Evolution exploration strategies (i.e., search algorithms) (#3377)
+
+Training service
+""""""""""""""""
+
+* Support shared storage for reuse mode (#3354)
+* Support Windows as the local training service in hybrid mode (#3353)
+* Remove PAIYarn training service (#3327)
+* Add "recently-idle" scheduling algorithm and set it as default scheduling algorithm in reuse mode (#3375)
+* Deprecated ``preCommand`` and enable ``pythonPath`` for remote training service (#3284 #3410)
```
Comment (J-shang) [inline docs/en_US/Release.rst:31]: Deprecated -> Deprecate

File: docs/en_US/Release.rst [hunk 9]
```diff
@@ -5,6 +5,70 @@
...
+
+nnictl & nni.experiment
+"""""""""""""""""""""""
+
+* Migrate nnicli to new Python API ``nni.experiment`` (#3334)
+* Refactor the way of specifying tuner in experiment Python API (\ ``nni.experiment``\ ), more aligned with nnictl (#3419)
```
Comment (J-shang) [inline docs/en_US/Release.rst:37]: ``nnicli``  ``nnictl``

File: docs/en_US/Release.rst (hunk 6 above)
Comment (J-shang) [reply docs/en_US/Release.rst:61]: -> training service

Refactor reuse mode temp folder.

File: docs/en_US/Release.rst [hunk 10]
```diff
@@ -5,6 +5,70 @@
...
+Release 2.1 - 3/10/2021
+-----------------------
+
+Major updates
+^^^^^^^^^^^^^
+
+Neural architecture search
+""""""""""""""""""""""""""
+
+* Improve NAS 2.0 (Retiarii) Framework (Alpha release)
```
Comment (J-shang) [inline docs/en_US/Release.rst:17]: rm `Alpha release`
add link:   https://github.com/microsoft/nni/issues/3301

File: docs/en_US/Release.rst [hunk 11]
```diff
@@ -5,6 +5,71 @@
...
+Release 2.1 - 3/10/2021
+-----------------------
+
+Major updates
+^^^^^^^^^^^^^
+
+Neural architecture search
+""""""""""""""""""""""""""
+
+* Improve NAS 2.0 (Retiarii) Framework (Imporved Experimental)
```
Comment (kvartet) [inline docs/en_US/Release.rst]: @QuanluZhang, please check the description of "Improved Experimental".

File: docs/en_US/Release.rst [hunk 12]
```diff
@@ -5,6 +5,71 @@
...
+Release 2.1 - 3/10/2021
+-----------------------
+
+Major updates
+^^^^^^^^^^^^^
+
+Neural architecture search
+""""""""""""""""""""""""""
+
+* Improve NAS 2.0 (Retiarii) Framework (Improved Experimental)
```
Comment (kvartet) [inline docs/en_US/Release.rst:17]: @QuanluZhang, please check the description of "Improved Experimental", thank you~

File: docs/en_US/Release.rst (hunk 1 above)
Comment (QuanluZhang) [reply docs/en_US/Release.rst:28]: it is fine for now

**Review by J-shang — APPROVED** [review]

**Pull request merged**